### PR TITLE
Allow witness use dev build of the rippled

### PR DESCRIPTION
**Pull request by oleks-rip**
Fix build process documentation
Add more checks to build system

### CODE REVIEW

File: Builds/CMake/deps/FindRipple.cmake [hunk 1]
```diff
@@ -0,0 +1,36 @@
+find_path(Ripple_SRC
+  NAMES src/ripple/protocol/STXChainBridge.h
+  PATHS ${RIPPLE_SRC_DIR}
+  NO_CMAKE_SYSTEM_PATH
+  NO_SYSTEM_ENVIRONMENT_PATH)
+if(Ripple_SRC)
+  set(Ripple_INCLUDE_DIR ${Ripple_SRC}/src)
+endif()
```
Comment (seelabs) [inline Builds/CMake/deps/FindRipple.cmake:8]: I'm not sure what's going on here. Why are we singling out `STXChainBridge.h`. To be clear, I'm not saying this is wrong, I really just don't understand.

Comment (oleks-rip) [reply Builds/CMake/deps/FindRipple.cmake:8]: We need to check that at least some files exists here.  STXChainBridge.h was chosen as it is only exist in xbridge branch (and will be existing after merge) so we just ensure that we are using correct ripple's src.

Comment (seelabs) [reply Builds/CMake/deps/FindRipple.cmake:8]: This seems fragile. Maybe there's a less fragile way to do this? If there's not I'll merge it, but I think we should at least try.

**Comment by seelabs** [comment]
merged

**Pull request closed without merging**